app/questionnaire/navigation.py

```diff
@@ -0,0 +1,151 @@
+import copy
+import logging
+
+from collections import defaultdict
+
+from app.data_model.answer_store import AnswerStore
+
+from app.helpers.schema_helper import SchemaHelper
+from app.questionnaire.location import Location
+from app.questionnaire.rules import evaluate_repeat
+
+logger = logging.getLogger(__name__)
+
+
+class Navigation(object):
+    """
+    Reads navigation config from the schema and returns collections of dicts that describe
+    completion status of each group
+    """
+    def __init__(self, survey_json, metadata=None, answer_store=None, completed_blocks=None):
```

Comment:
answer_store should be a mandatory argument. When would we ever use navigation without taking into account the users answers similarly for completed blocks. We are just making more scenarios we should have unit tests for by making things optional just because we can